src/DynamoCoreWpf/UI/Converters.cs

```diff
@@ -2341,4 +2349,292 @@ public object ConvertBack(object value, Type targetType, object parameter, Cultu
+                    }
+
+                    //If it is root category, then move the vertical line outside the grid.
+                    if (childMargin.Left == parentMargin.Left)
+                    {                      
+                        return new Thickness(-10, 0, 0, 0);
+                    }
+                   
+                    //For levels 0,1,2, the difference will be less. 
+                    //For deep levels, the expander left margin will be increased by 20. 
+                    //Hence the difference will be greater.
+                    if (diff < childMargin.Right)
+                    {
+                        return new Thickness(0, 0, childMargin.Left * 2, 0);
+                    }
+
+                    return new Thickness(diff, 0, diff * 2, 0);
+                }
+
+                //Default. Move the line outside the grid.
+                return new Thickness(-10, 0, 0, 0);
+            }
+
+            public object[] ConvertBack(object value, Type[] targetTypes, object parameter, CultureInfo culture)
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        /// <summary>
+        /// This controls the horizontal line margin
+        /// </summary>
+        public class TreeViewHLineMarginConverter : IMultiValueConverter
+        {           
+            public object Convert(object[] values, Type targetType, object parameter, CultureInfo culture)
+            {
+                var VerLnMargin = (Thickness)(values[0]);
+                var expanderMargin = (Thickness)(values[1]);
+                              
+                //Find if the item is last
+                var item = (TreeViewItem) values[2];
+                ItemsControl ic = ItemsControl.ItemsControlFromItemContainer(item);
+                var level = -1;
+                var isLastItem = ic.ItemContainerGenerator.IndexFromContainer(item) == ic.Items.Count - 1;
+                if (values[2] is DependencyObject)
+                {
+                    var parent = VisualTreeHelper.GetParent(values[2] as DependencyObject);
+                    bool gotParentTree = false;
+                    while (!(gotParentTree) && (parent != null))
+                    {
+                        if (parent is TreeViewItem)
+                            level++;
+                        parent = VisualTreeHelper.GetParent(parent);
+                        if (parent is System.Windows.Controls.TreeView)
+                        {
+                            var view = parent as System.Windows.Controls.TreeView;
+                            if (view.Name == "CategoryTreeView")
+                            {
+                                gotParentTree = true;
+                            }
+                        }
+                    }
+                }
+
+                var left = VerLnMargin.Left + 10;
+                var right = (expanderMargin.Right*2) + 5;
```

Comment:
Please replace all magic numbers with `const` fields - generally there should be no integers in line in your code.